core/commands/dht.go

```diff
@@ -581,3 +589,13 @@ PutValue will store the given key value pair in the dht.
 	},
 	Type: notif.QueryEvent{},
 }
+
+func escapeDhtKey(s string) key.Key {
+	parts := strings.Split(s, "/")
+	if len(parts) < 3 {
+		return key.B58KeyDecode(s)
+	}
```

Comment:
the key is odd, which wouldnt work for the path parsing and our path parsing makes sure the leading element is `ipfs` or `ipns`. I think the switch LGTM, i'll do that.